pages/assets/js/src/profile-pages/pg-dashboard.ts

```diff
@@ -158,4 +170,66 @@ export class PgDashboard extends OpenElement {
       </div>
     `;
   }
+
+  private async getLocationFromIP() {
+    const saved_location = localStorage.getItem("user_location");
+    this.user.location = saved_location ? JSON.parse(saved_location) : null;
+    if (this.user.location?.hash) {
+      this.user.location_hash = this.user.location.hash;
+    }
+
+    if (
+      !this.user.location ||
+      (this.user.location.date_set &&
+        this.user.location.date_set <
+        Date.now() - 604800000) /*7 days in milliseconds*/
+      ) {
+        await window.api_fetch(`https://geo.prayer.global/json`, {
+          method: "GET",
+        })
+        .then((response:any) => {
+          if (response) {
+            const locationData = {
+              lat: response.location.latitude,
+              lng: response.location.longitude,
+              label: `${response.city?.names?.en}, ${response.country?.names?.en}`,
+              country: response.country?.names?.en,
+              date_set: Date.now(),
+              source: "ip",
+            };
+            // Update the user's location
+            this.user.location = locationData;
+            
+            let pg_user_hash = localStorage.getItem("pg_user_hash");
+            if (!pg_user_hash || pg_user_hash === "undefined") {
+              pg_user_hash = window.crypto.randomUUID();
+              localStorage.setItem("pg_user_hash", pg_user_hash);
+              this.user.location_hash = pg_user_hash;
+            }
+            this.user.location.hash = pg_user_hash;
+            
+            localStorage.setItem("user_location", JSON.stringify(this.user.location));
+          }
+        });
+    }
+
+    await window.api_fetch(`${window.pg_global.root}pg-api/v1/dashboard/save_location`, {
+      method: "POST",
+      body: JSON.stringify({
+        location_hash: this.user.location_hash,
+        location: this.user.location,
+      }),
+    });
+    this.requestUpdate();
+  }
+
+  private async link_anonymous_prayers() {
+    const url = `${window.pg_global.root}pg-api/v1/dashboard/link_anonymous_prayers`;
+    window.api_fetch(url, {
+      method: "POST",
+      body: JSON.stringify({
+        location_hash: this.user.location_hash,
```

Comment:
is the location_hash to do with their location? or to do with the user?
we want to link anonymous prayers to the user independent of the location they are in (I'm assuming that is what is happening here, but just checking as the variable suggests it's to do with location)